src/Jackalope/Item.php

```diff
@@ -636,10 +636,6 @@ protected function checkState()
             $this->postDirtyState = -1;
         }
 
-        if ($this->state === self::STATE_DELETED) {
-            throw new InvalidItemStateException('Item '.$this->path.' is deleted');
-        }
```

Comment:
if you call refresh repeatedly, things will become very expensive as everything that is dirty is reloaded.

it smells fishy when a node is still dirty after a save() operations, btw. if that is the case there might be some other bug too.